examples/equilibria/fixed_boundary.ex.py

```diff
@@ -0,0 +1,137 @@
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Fixed boundary equilibrium example
+"""
+
+# %% [markdown]
+#
+# # Fixed boundary equilibrium example
+#
+# Imports
+
+# %%
+from datetime import datetime
+
+import dolfin
+import matplotlib.pyplot as plt
+
+from bluemira.base.components import PhysicalComponent
+from bluemira.equilibria.fem_fixed_boundary.fem_magnetostatic_2D import (
+    FemGradShafranovFixedBoundary,
+)
+from bluemira.equilibria.fem_fixed_boundary.file import save_fixed_boundary_to_file
+from bluemira.equilibria.fem_fixed_boundary.utilities import create_mesh
+from bluemira.equilibria.profiles import DoublePowerFunc, LaoPolynomialFunc
+from bluemira.equilibria.shapes import KuiroukidisLCFS
+from bluemira.geometry.face import BluemiraFace
+
+# %% [markdown]
+# Define some important values
+
+# %%
+R_0 = 9  # [m]
+I_p = 17e6  # [A]
+B_0 = 5  # [T]
+
+# %% [markdown]
+# Let's define a boundary shape for the fixed boundary equilibrium
+
+# %%
+parameterisation = KuiroukidisLCFS(
+    {
+        "kappa_u": {"value": 1.7},
+        "kappa_l": {"value": 1.8},
+        "delta_u": {"value": 0.33},
+        "delta_l": {"value": 0.4},
+    }
+)
+
+lcfs_shape = parameterisation.create_shape("LCFS", n_points=100)
+lcfs_face = BluemiraFace(lcfs_shape)
+
+# %% [markdown]
+# Next we need to mesh this geometry
+
+# %%
+plasma = PhysicalComponent("plasma", lcfs_face)
+plasma.shape.mesh_options = {"lcar": 0.3, "physical_group": "plasma_face"}
+plasma.shape.boundary[0].mesh_options = {"lcar": 0.3, "physical_group": "lcfs"}
+
+mesh = create_mesh(plasma, ".", "fixed_boundary_example", "fixed_boundary_example.msh")
+
+dolfin.plot(mesh)
+plt.show()
+# %% [markdown]
+# Now we define some profile functions for p' and FF'.
+# We'll use some typical functional forms for this, but you are free to specify
+# the flux functions using whichever callable you like.
+
+# %%
+
+p_prime = LaoPolynomialFunc([2, 3, 1])
+ff_prime = DoublePowerFunc([1.5, 2])
+
+
+# %% [markdown]
+# Set up the esolver and run it
+
+# %%
+
+solver = FemGradShafranovFixedBoundary(
+    I_p=I_p, R_0=R_0, B_0=B_0, p_order=2, max_iter=30, iter_err_max=1e-4, relaxation=0.05
+)
+solver.set_mesh(mesh)
+solver.set_profiles(p_prime, ff_prime)
+equilibrium = solver.solve(plot=True)
+
+
+# %% [markdown]
+# Let's inspect the result
+
```

Comment:
This is less trivial than I would like... will till the day I die be a fan of `something.plot()` :)